### PR TITLE
Cpu Contiguous Dim Check

### PR DESCRIPTION
**Pull request by kelvinxu**
Adds a dimension check pull request at #2949

### CODE REVIEW

File: theano/tensor/extra_ops.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def c_code(self, node, name, inames, onames, sub):
             if (!PyArray_CHKFLAGS(%(x)s, NPY_ARRAY_C_CONTIGUOUS)){
                 // check to see if output is contiguous first
                 if (%(y)s != NULL &&
+                    PyArray_CompareLists(PyArray_DIMS(%(y)s), PyArray_DIMS(%(x)s), PyArray_NDIMS(%(x)s)) &&
```
Comment (nouiz) [inline theano/tensor/extra_ops.py]: You need to bump c_code_cache_version.

**Comment by nouiz** [comment]
we can merge when travis pass.

**Comment by lamblin** [comment]
This is failing with an error saying: `'PyArray_NDIMS' was not declared in this scope`.
I did not check why that would happen yet.

**Comment by nouiz** [comment]
I think it should be PyArray_NDIMS

**Comment by nouiz** [comment]
I think it should be PyArray_NDIM

**Comment by kelvinxu** [comment]
Amended

**Pull request merged**